### ISSUE TITLE
YADCF and column visibility issue "ERROR: Filter container could not be found." / select2 warnings

### ISSUE THREAD

**Issue by chimos** [vedmack/yadcf#357]
Hi, 
I already used yadcf with Buttons-Colvis together without any issue, but after some changes yadcf shows this error message in the console: _ERROR: Filter container could not be found_, only when some columns are hidden after yadcf init.

See [test fiddle](https://jsfiddle.net/chimos/1ojk5aqg/30/)

I'm hidding some columns in document ready as a default and it seems to be causing the issue. Checked commenting these lines out:

```
    // Default hidden columns
    my_table.columns(':gt(8):not(".filtered_col")').visible(false, false);
    my_table.columns.adjust().draw(false); // adjust column sizing and redraw

```

Any idea? Thank you

**Comment by vedmack** [comment]
Hi,
Please clean your example a bit, you trying to access filters with index 9+ while there are no columns 9+ defined, see a bit cleaned test page https://jsfiddle.net/1ojk5aqg/32/ (no errors in console)

**Comment by chimos** [comment]
Thanks for your answer. And sorry for the unnecessary complexity, I simplified  a bit. Also I forgot some filters in the first test (copy-paste error). Now all filters should be correctly initialised but I keep getting that error log.

See [updated fiddle](https://jsfiddle.net/chimos/1ojk5aqg/38/).

**Comment by vedmack** [comment]
I pointed you jsfiddle to my local version of yadcf which got better logging, take a [look at the test page](https://jsfiddle.net/1ojk5aqg/39/) and inspect the console of the dev tools, I think it will explain you the reason behind the error (trying to access columns/filter containers that do not exist)

**Comment by chimos** [comment]
I understand, external containers didn't exist at that point.

Now I am using the "default-direct-in-header" filters and moving `.yadcf-filter-wrapper` inside the dropdown in **draw callback**, this way:
```javascript
          //(inside DataTables init ...)
          'drawCallback': function(settings) {
            var tar_th = $('#example > thead').find('th');
            $(tar_th).each(function() {
              var submenu = $(this).find('.submenu');
              $(this).find('> .yadcf-filter-wrapper').appendTo(submenu);
            });
          },
         //(...)
```
And, since I am using `column-visibility` together with `StateSave`, I need to do the same each time the end user sets a hidden column to visible, this way:
```javascript
        my_table.on( 'buttons-action', function ( e, buttonApi, dataTable, node, config ) {
          if ( buttonApi.node().hasClass('buttons-columnVisibility active') ) {
            var tar_th = $('#example > thead').find('th');
            $(tar_th).each(function() {
              var submenu = $(this).find('.submenu');
              $(this).find('> .yadcf-filter-wrapper').appendTo(submenu);
            });
          }
        });
```
It took me a while to put this together, because of my poor coding knowledge and because in my local environment I was using version 0.8.x and this code works only if I use 0.9.x of YADCF.

BUT, with 0.9.x came another issue that is apparently not ther with 0.8.x, when the end user selects something in a Select2 filter:

> VM2179 select2.min.js:21 Uncaught TypeError: Cannot read property '0' of null

see [updated fiddle](https://jsfiddle.net/chimos/y5fkwg6j/8/) (and select some value in a Select2 filter to reproduce the error)

Please tell me if I should open a new issue for that or a question in stackoverflow. 

Thank you!

**Comment by vedmack** [comment]
You are right, with [yadcf 0.8.3](https://jsfiddle.net/y5fkwg6j/12/) it didn't throw anything to the console, It doesn't look harmful (google is full of this select2 error) I'll try to see if its possible to fix it, I had to change the way I use select2 because otherwise it does not work as expected in other scenarios. ( I'll keep this issue open for now - no need for new one)

**Comment by vedmack** [comment]
select2 warnings were fixed in [0.9.2.beta.1](https://github.com/vedmack/yadcf/blob/3db921a674f785af22d436b2676e8bf04c07004a/src/jquery.dataTables.yadcf.js)
